lib/src/http/utils.dart

```diff
@@ -49,11 +53,15 @@ Duration calculateBackOff(RequestAutoRetry autoRetry,
 }
 
 Duration _calculateExponentialBackOff(RequestAutoRetry autoRetry,
-    {@visibleForTesting Random random}) {
+    {@visibleForTesting Random random, AdvancedBackOffCalculator calculator}) {
   if (autoRetry.backOff.withJitter ?? false) {
-    final jitteredBackOff = _calculateAdvancedExponentialJitteredBackOffInMs(
-        autoRetry,
-        random: random);
+    if (calculator == null) {
+      throw StateError(
+          'Cannot calculate advanced jitter without an instance of AdvancedBackOffCalculator');
+    }
```

Comment:
Why not just make `calculator` a required param?

Reply:
Because you can have backoff without jitter